pkg/cmd/package/support/pack.go

```diff
@@ -0,0 +1,270 @@
+	Ask      question.Asker
+	NoPrompt bool
+	CmdPath  string
+}
+
+func NewPackageCreateFlags() *PackageCreateFlags {
+	return &PackageCreateFlags{
+		Id:        flag.New[string](FlagId, false),
+		Version:   flag.New[string](FlagVersion, false),
+		BasePath:  flag.New[string](FlagBasePath, false),
+		OutFolder: flag.New[string](FlagOutFolder, false),
+		Include:   flag.New[[]string](FlagInclude, false),
+		Verbose:   flag.New[bool](FlagVerbose, false),
+		Overwrite: flag.New[bool](FlagOverwrite, false),
+	}
+}
+
+func NewPackageCreateOptions(f factory.Factory, flags *PackageCreateFlags, cmd *cobra.Command) *PackageCreateOptions {
+	return &PackageCreateOptions{
+		PackageCreateFlags: flags,
+		Writer:             cmd.OutOrStdout(),
+		Ask:                f.Ask,
+		NoPrompt:           !f.IsPromptEnabled(),
+		CmdPath:            cmd.CommandPath(),
+	}
+}
+
+func PackageCreatePromptMissing(opts *PackageCreateOptions) error {
+	if opts.Id.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Package ID",
+			Help:    "The ID of the package.",
+		}, &opts.Id.Value, survey.WithValidator(survey.ComposeValidators(
+			survey.Required,
+			survey.MaxLength(200),
+			survey.MinLength(1),
+		))); err != nil {
+			return err
+		}
+	}
+
+	if opts.Version.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Version",
+			Help:    "The version of the package, must be a valid SemVer.",
+		}, &opts.Version.Value); err != nil {
+			return err
+		}
+	}
+
+	if opts.BasePath.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Base Path",
+			Help:    "Root folder containing the contents to zip; defaults to current directory.",
+			Default: ".",
+		}, &opts.BasePath.Value); err != nil {
+			return err
+		}
+	}
+
+	if opts.OutFolder.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Out Folder",
+			Help:    "Folder into which the zip file will be written; defaults to the current directory.",
+			Default: ".",
+		}, &opts.OutFolder.Value); err != nil {
+			return err
+		}
+	}
+
+	if len(opts.Include.Value) == 0 {
+		for {
+			var pattern string
+			if err := opts.Ask(&survey.Input{
+				Message: "Include patterns",
+				Help:    "Add a file pattern to include, relative to the base path e.g. /bin/*.dll; defaults to \"**\" no patterns provided.",
+			}, &pattern); err != nil {
+				return err
+			}
+			if pattern == "" {
+				break
+			}
+			opts.Include.Value = append(opts.Include.Value, pattern)
+		}
+	}
+
+	if !opts.Verbose.Value {
+		err := opts.Ask(&survey.Confirm{
+			Message: "Verbose",
+			Default: true,
+		}, &opts.Verbose.Value)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !opts.Overwrite.Value {
+		err := opts.Ask(&survey.Confirm{
+			Message: "Overwrite",
+			Default: true,
+		}, &opts.Overwrite.Value)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func VerboseOut(isVerbose bool, messageTemplate string, messageArgs ...any) {
+	if isVerbose {
+		fmt.Printf(messageTemplate, messageArgs...)
```

Comment:
IIRC we want to run message printing using the cobra cmd.print functions or other redirected IO; if you just do bare printf then things like color escape codes are broken on windows and NO_COLOR doesn't work